airflow/jobs/scheduler_job_runner.py

```diff
@@ -1831,3 +1836,16 @@ def _orphan_unreferenced_datasets(self, session: Session = NEW_SESSION) -> None:
 
         updated_count = sum(self._set_orphaned(dataset) for dataset in orphaned_dataset_query)
         Stats.gauge("dataset.orphaned", updated_count)
+
+    def _executor_to_tis(self, tis: list[TaskInstance]) -> dict[BaseExecutor, list[TaskInstance]]:
+        """Organize TIs into lists per their respective executor."""
+        _executor_to_tis: defaultdict[BaseExecutor, list[TaskInstance]] = defaultdict(list)
+        executor: str | None
+        for ti in tis:
+            if ti.executor:
+                executor = str(ti.executor)
+            else:
+                executor = None
```

Comment:
This bit of weirdness is just to make mypy types happy. The ti.executor field type comes up as a SQLA Column type, so if it's not None I cast to a string and if it's None I just make it None again (because mypy is stupid and harms my life).